requirements-dev.txt

```
@@ -14,6 +14,7 @@ clang-tidy~=16.0
 clang-format~=16.0
 isort==5.13.2
 click==8.0.4
-cmake==3.28.4
+# cmake 3.29.0 does not work with python <3.10
+cmake!=3.29.0
```

I would leave this untouched as all other deps in this file are pinned to specific versions.
```suggestion
cmake==3.28.4
```